creator-node/test/CNodeHealthManager.test.js

```diff
@@ -0,0 +1,478 @@
+
+      // Verify that only the simple check was performed and returned healthy
+      const isHealthy = await CNodeHealthManager.isNodeHealthy(
+        node,
+        performSimpleCheck
+      )
+      expect(isHealthy).to.be.false
+      expect(queryVerboseHealthCheckStub).to.have.been.calledOnceWithExactly(
+        node
+      )
+      expect(determinePeerHealthStub).to.not.have.been.called
+      expect(logErrorStub).to.have.been.called.calledOnceWithExactly(
+        `isNodeHealthy() peer=${node} is unhealthy: ${error.toString()}`
+      )
+    })
+  }
+
+  it('returns false when determinePeerHealth throws with performSimpleCheck=false', async function () {
+    // Stub functions that isNodeHealthy() will call
+    const node = 'http://some_content_node.co'
+    const determinePeerHealthError = new Error('test determinePeerHealthError')
+    const verboseHealthCheckResp = { healthy: false }
+    const queryVerboseHealthCheckStub = sandbox
+      .stub(CNodeHealthManager, 'queryVerboseHealthCheck')
+      .resolves(verboseHealthCheckResp)
+    const determinePeerHealthStub = sandbox
+      .stub(CNodeHealthManager, 'determinePeerHealth')
+      .throws(determinePeerHealthError)
+    const logErrorStub = sandbox.stub(CNodeHealthManager, 'logError')
+
+    // Verify that only the simple check was performed and returned healthy
+    const isHealthy = await CNodeHealthManager.isNodeHealthy(node, false)
+    expect(isHealthy).to.be.false
+    expect(queryVerboseHealthCheckStub).to.have.been.calledOnceWithExactly(node)
+    expect(determinePeerHealthStub).to.have.been.calledOnceWithExactly(
+      verboseHealthCheckResp
+    )
+    expect(logErrorStub).to.have.been.called.calledOnceWithExactly(
+      `isNodeHealthy() peer=${node} is unhealthy: ${determinePeerHealthError.toString()}`
+    )
+  })
+
+  it("returns true when determinePeerHealth doesn't throw with performSimpleCheck=false", async function () {
+    // Stub functions that isNodeHealthy() will call
+    const node = 'http://some_content_node.co'
+    const verboseHealthCheckResp = { healthy: true }
+    const queryVerboseHealthCheckStub = sandbox
+      .stub(CNodeHealthManager, 'queryVerboseHealthCheck')
+      .resolves(verboseHealthCheckResp)
+    const determinePeerHealthStub = sandbox.stub(
+      CNodeHealthManager,
+      'determinePeerHealth'
+    )
+    const logErrorStub = sandbox.stub(CNodeHealthManager, 'logError')
+
+    // Verify that only the simple check was performed and returned healthy
+    const isHealthy = await CNodeHealthManager.isNodeHealthy(node, false)
+    expect(isHealthy).to.be.true
+    expect(queryVerboseHealthCheckStub).to.have.been.calledOnceWithExactly(node)
+    expect(determinePeerHealthStub).to.have.been.calledOnceWithExactly(
+      verboseHealthCheckResp
+    )
+    expect(logErrorStub).to.not.have.been.called.called
+  })
+
+  it("returns true when determinePeerHealth doesn't throw with default performSimpleCheck", async function () {
```

Comment:
I think it wouldve been ok to not have this test, since this test is more so testing whether js defaulting works. and since you wrote a test that covers the `performSimpleCheck=false` case already, this prob isn't necessary :)

Reply:
it definitely felt weird to write this test and you're right. the reason I did it is because it was the only branch that was missing for 100% coverage of the file lol

Reply:
👍